pkg/xds/generator/outbound_proxy_generator.go

```diff
@@ -315,22 +318,36 @@ func (_ OutboundProxyGenerator) determineRoutes(proxy *model.Proxy, outbound *ku
 		return clusters
 	}
 
-	for _, http := range route.Spec.GetConf().GetHttp() {
-		route := envoy_common.Route{
-			Match:    http.Match,
-			Modify:   http.Modify,
-			Clusters: clustersFromSplit(http.GetSplitWithDestination()),
+	addRoutes := func(route *mesh_core.TrafficRouteResource, rateLimit *mesh_proto.RateLimit, routes envoy_common.Routes) envoy_common.Routes {
+		for _, http := range route.Spec.GetConf().GetHttp() {
+			route := envoy_common.Route{
+				Match:     http.Match,
+				Modify:    http.Modify,
+				RateLimit: rateLimit,
+				Clusters:  clustersFromSplit(http.GetSplitWithDestination()),
+			}
+			routes = append(routes, route)
 		}
-		routes = append(routes, route)
+
+		if defaultDestination := route.Spec.GetConf().GetSplitWithDestination(); len(defaultDestination) > 0 {
+			cfs := clustersFromSplit(defaultDestination)
+			route := envoy_common.Route{
+				Match:     nil,
+				Clusters:  cfs,
+				RateLimit: rateLimit,
+			}
+			routes = append(routes, route)
+		}
+		return routes
 	}
 
-	if defaultDestination := route.Spec.GetConf().GetSplitWithDestination(); len(defaultDestination) > 0 {
-		cfs := clustersFromSplit(defaultDestination)
-		route := envoy_common.Route{
-			Match:    nil,
-			Clusters: cfs,
+	if len(proxy.Policies.RateLimits.Outbound[oface]) < 1 {
+		routes = addRoutes(route, nil, routes)
+	} else {
+		for _, rateLimit := range proxy.Policies.RateLimits.Outbound[oface] {
+			routes = addRoutes(route, rateLimit, routes)
 		}
-		routes = append(routes, route)
 	}
+
```

Comment:
While we are here, could we remove the named return values?

Reply:
Is there any specific reason to remove it?

To be honest, I'm not a fan of "named return values", but I can see how it can save lines etc.

Right now we have number of occurrences of named return values across the project. What would be the rule when we can use it and when we avoid it?